speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DatabaseSettingsMenu.cpp

```diff
@@ -0,0 +1,264 @@
+#include <tgfclient.h>
+#include "guimenu.h"
+#include "legacymenu.h"
+#include "Mediator.h"
+#include "DatabaseSettingsMenu.h"
+#include "DatabaseConnectionCheck.h"
+#include "DataSelectionMenu.h"
+#include "../rppUtils/FileDialog.hpp"
+
+#include <experimental/filesystem>
+
+// GUI screen handles
+static void* s_scrHandle = nullptr;
+static void* s_nextHandle = nullptr;
+
+// Control variables for the menu
+int m_usernameControl;
+int m_passwordControl;
+int m_addressControl;
+int m_portControl;
+int m_schemaControl;
+int m_useSSLControl;
+
+char m_portString[256];
+
+int m_caCertFileDialogControl;
+int m_publicCertFileDialogControl;
+int m_privateCertFileDialogControl;
+int m_caCertDialogLabel;
+int m_publicCertDialogLabel;
+int m_privateCertDialogLabel;
+int m_dbStatusControl;
+bool m_certChosen = false;
+
+tDatabaseSettings m_dbsettings;
+
+/// @brief Handle input in the Username textbox
+static void SetUsername(void*)
+{
+    strcpy_s(m_dbsettings.Username, SETTINGS_NAME_LENGTH, GfuiEditboxGetString(s_scrHandle, m_usernameControl));
+    GfuiEditboxSetString(s_scrHandle, m_usernameControl, m_dbsettings.Username);
+}
+
+/// @brief Handle input in the Password textbox
+static void SetPassword(void*)
+{
+    strcpy_s(m_dbsettings.Password, SETTINGS_NAME_LENGTH, GfuiEditboxGetString(s_scrHandle, m_passwordControl));
+    GfuiEditboxSetString(s_scrHandle, m_passwordControl, m_dbsettings.Password);
+}
+
+/// @brief Handle input in the Address textbox
+static void SetAddress(void*)
+{
+    strcpy_s(m_dbsettings.Address, SETTINGS_NAME_LENGTH, GfuiEditboxGetString(s_scrHandle, m_addressControl));
+    GfuiEditboxSetString(s_scrHandle, m_addressControl, m_dbsettings.Address);
+}
+
+/// @brief Handle input in the Port textbox
+static void SetPort(void*)
+{
+    sprintf(m_portString, GfuiEditboxGetString(s_scrHandle, m_portControl));
```

Comment:
Should use strcpy as well (does this need a strcpy? Since the string is not what you want saved.)

Reply:
I wasn't able to do it another way but if possible only saving it as an integer is indeed preferable